woodwork/accessor_utils.py

```diff
@@ -118,11 +118,7 @@ def _get_valid_dtype(series, logical_type):
     with the given logical_type"""
```

Comment:
Yeah, I get your point here. Maybe it would be best to handle this as a separate dtype cleanup Issue/PR. I'm thinking it might be cleanest to keep the logic inside this function of whether the dtype should be the panads_dtype or the backup_dtype, otherwise we could potentially be sprinkling that logic at various points in the code (to get the value of `needs_backup`). We could maybe update this to take the series type and the logical type instead of passing the series, and maybe that would make more sense? Anyway, we can discuss more and settle on a something better than we currently have.

We could also consider the naming (pandas_dtype/backup_dtype) as part of that cleanup. It bothers me a little bit that we use pandas_dtype for both pandas and dask, but we only use backup_dtype for koalas.

Reply:
I think passing `type(series)` to `_get_valid_dtype` makes sense as well as having a separate dtype cleanup issue. I'll create that issue and add all the parts of the cleanup that I can think of and also close #670 since anything we were doing there should be covered by the cleanup issue.

Reply:
Added #681